src/witan/app/forecast.clj

```diff
@@ -459,16 +463,17 @@
   :allowed-methods #{:post}
   :processable? (fn [ctx]
                   (let [forecast (get-most-recent-version id)
-                        inputs (:inputs (util/get-post-params ctx))]
-                    (and forecast
-                         ((util/post!-processable-validation ws/UpdateForecast) ctx)
-                         (all-categories-exist-in-model? forecast (keys inputs))
-                         (every? data/exists? inputs)
-                         (or (-> forecast :public? not) ;; if public, check all data inputs are public
-                             (every? #(-> % second :public?) inputs)))))
+                        inputs   (:inputs (util/get-post-params ctx))
+                        tests    [[(fn [] forecast)                                                    "forecast is nil"]
```

Comment:
For this number of tests I don't think the complexity is warranted. If the tests were configurable in some way then we'd want to do something programmically along these lines, but since it's two harded checks, one of which is a nil? check it seems pragmatic to use an if ladder.
```
(if forecast
       (if-not ((util/post!-processable-validation ws/UpdateForecast) ctx)
            (log/error ...))
       (log/error ...))
```